### PR TITLE
Api work

### PR DESCRIPTION
**Pull request by Nitro2000**
Api Work branch to main

### CODE REVIEW

File: app/src/main/java/com/example/teamhiring/presentation/adapters/AllJobAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        val jobTitle: TextView = itemView.findViewById(R.id.jobTitleTxt)
+        val jobQualification: TextView = itemView.findViewById(R.id.jobHighQualTxt)
+        val jobExpSubTxt: TextView = itemView.findViewById(R.id.jobExpSubTxt)
+        val jobAppAnnSalTxt: TextView = itemView.findViewById(R.id.jobAppAnnSalTxt)
+        val jobTypeTxt: TextView = itemView.findViewById(R.id.jobTypeTxt)
+        val jobTimeTypeTxt: TextView = itemView.findViewById(R.id.jobTimeTypeTxt)
+        val jobCompanyTxt: TextView = itemView.findViewById(R.id.jobExpCompTxt)
+        val jobRecNameTxt: TextView = itemView.findViewById(R.id.jobExpDesignTxt)
+        val jobAppNameTxt: TextView = itemView.findViewById(R.id.jobAppNameTxt)
+        val jobEmpDescTxt: TextView = itemView.findViewById(R.id.jobEmpDescTxt)
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/AllJobAdapter.kt:27]: Use binding don't find id, its already in binding

File: app/src/main/java/com/example/teamhiring/data/repositories/EmplyeeSavedJobRepository.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import com.example.teamhiring.data.models.SavedJobData
+import com.example.teamhiring.data.services.ReferenceApiService
+import retrofit2.Response
+import javax.inject.Inject
+
+class EmplyeeSavedJobRepository @Inject constructor(private val referenceApiService: ReferenceApiService) {
+    suspend fun getSavedJob(): Response<List<SavedJobData>> {
+        return referenceApiService.getSavedJobList()
+    }
+}
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/data/repositories/EmplyeeSavedJobRepository.kt:12]: No need for all this repository, if api service is same,  Make one repository file for job data, all job data saved, applied, viewed by

File: app/src/main/java/com/example/teamhiring/data/services/ReferenceApiService.kt [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
+
+    @GET("appliedjob?emp_id=5")
+    suspend fun getAppliedJobList(): Response<List<AppliedJobData>>
+
+    @GET("saved_jobs?emp_id=1")
+    suspend fun getSavedJobList(): Response<List<SavedJobData>>
+
+    @GET("saved_jobs?emp_id=1")
+    suspend fun getViewedByList(): Response<List<EmpAppViewedData>>
 }
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/data/services/ReferenceApiService.kt:23]: This api service is just for reference tutorial, make a new api service with appropriate name or change this api service name

File: app/src/main/java/com/example/teamhiring/presentation/adapters/AllJobAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    override fun getItemCount(): Int = jobList.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val ItemJobModel = jobList[position]
+
+        holder.jobTitle.text = ItemJobModel.jobTitle
+        holder.jobCompanyTxt.text = ItemJobModel.jobTitle
+        holder.jobAppAnnSalTxt.text = "Rs. "+ItemJobModel.jobExMinSalary +"-"+ItemJobModel.jobExMaxSalary +" LPA"
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/AllJobAdapter.kt:48]: Try not to use hardcoded string directly in kotlin files, use  context.getString(R.string.whatever)  it also provide f strings

File: app/src/main/java/com/example/teamhiring/presentation/adapters/EmpAppViewedAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        val jobViewedTitle: TextView = itemView.findViewById(R.id.jobTitleTxt)
+        val jobViewedQualification: TextView = itemView.findViewById(R.id.jobHighQualTxt)
+        val jobViewedExpSubTxt: TextView = itemView.findViewById(R.id.jobExpSubTxt)
+        val jobViewedAppAnnSalTxt: TextView = itemView.findViewById(R.id.jobAppAnnSalTxt)
+        val jobViewedTypeTxt: TextView = itemView.findViewById(R.id.jobTypeTxt)
+        val jobViewedTimeTypeTxt: TextView = itemView.findViewById(R.id.jobTimeTypeTxt)
+        val jobViewedCompanyTxt: TextView = itemView.findViewById(R.id.jobExpCompTxt)
+        val jobViewedRecNameTxt: TextView = itemView.findViewById(R.id.jobExpDesignTxt)
+        val jobViewedAppNameTxt: TextView = itemView.findViewById(R.id.jobAppNameTxt)
+        val jobViewedEmpDescTxt: TextView = itemView.findViewById(R.id.jobEmpDescTxt)
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/EmpAppViewedAdapter.kt:27]: same use binding

File: app/src/main/java/com/example/teamhiring/presentation/adapters/EmpAppViewedAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+        holder.jobViewedTitle.text = ItemViewedByModel.jobViewedTitle
+        holder.jobViewedCompanyTxt.text = ItemViewedByModel.jobViewedTitle
+        holder.jobViewedAppAnnSalTxt.text = "Rs. "+ItemViewedByModel.jobViewedExMinSalary +"-"+ItemViewedByModel.jobViewedExMaxSalary +" LPA"
+        holder.jobViewedEmpDescTxt.text = ItemViewedByModel.jobViewedDescription
+        holder.jobViewedQualification.text =ItemViewedByModel.jobViewedQualification
+        holder.jobViewedTypeTxt.text = ItemViewedByModel.jobViewedEmpType
+        holder.jobViewedExpSubTxt.text = ItemViewedByModel.jobViewedExperience
+        holder.jobViewedTimeTypeTxt.text = ItemViewedByModel.jobViewedJobType
+        holder.jobViewedExpSubTxt.text = ItemViewedByModel.jobViewedExperience
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/EmpAppViewedAdapter.kt:55]: use binding

File: app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeAppliedJobAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        val jobAppliedTitle: TextView = itemView.findViewById(R.id.jobTitleTxt)
+        val jobAppliedQualification: TextView = itemView.findViewById(R.id.jobHighQualTxt)
+        val jobAppliedExpSubTxt: TextView = itemView.findViewById(R.id.jobExpSubTxt)
+        val jobAppliedAppAnnSalTxt: TextView = itemView.findViewById(R.id.jobAppAnnSalTxt)
+        val jobAppliedTypeTxt: TextView = itemView.findViewById(R.id.jobTypeTxt)
+        val jobAppliedTimeTypeTxt: TextView = itemView.findViewById(R.id.jobTimeTypeTxt)
+        val jobAppliedCompanyTxt: TextView = itemView.findViewById(R.id.jobExpCompTxt)
+        val jobAppliedRecNameTxt: TextView = itemView.findViewById(R.id.jobExpDesignTxt)
+        val jobAppliedAppNameTxt: TextView = itemView.findViewById(R.id.jobAppNameTxt)
+        val jobAppliedEmpDescTxt: TextView = itemView.findViewById(R.id.jobEmpDescTxt)
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeAppliedJobAdapter.kt:25]: use binding

File: app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeAppliedJobAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        var ItemAppliedJob = appliedJobList[position]
+        holder.jobAppliedTitle.text = ItemAppliedJob.jobAppliedTitle
+        holder.jobAppliedCompanyTxt.text = ItemAppliedJob.jobAppliedTitle
+        holder.jobAppliedAppAnnSalTxt.text = "Rs. "+ItemAppliedJob.jobAppliedExMinSalary +"-"+ItemAppliedJob.jobAppliedExMaxSalary
+        holder.jobAppliedEmpDescTxt.text = ItemAppliedJob.jobAppliedDescription
+        holder.jobAppliedQualification.text =ItemAppliedJob.jobAppliedQualification
+        holder.jobAppliedTypeTxt.text = ItemAppliedJob.jobAppliedEmpType
+        holder.jobAppliedExpSubTxt.text = ItemAppliedJob.jobAppliedExperience
+        holder.jobAppliedTimeTypeTxt.text = ItemAppliedJob.jobAppliedJobType
+        holder.jobAppliedExpSubTxt.text = ItemAppliedJob.jobAppliedExperience
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeAppliedJobAdapter.kt:51]: same

File: app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeSavedJobAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        val jobSavedTitle: TextView = itemView.findViewById(R.id.jobTitleTxt)
+        val jobSavedQualification: TextView = itemView.findViewById(R.id.jobHighQualTxt)
+        val jobSavedExpSubTxt: TextView = itemView.findViewById(R.id.jobExpSubTxt)
+        val jobSavedAppAnnSalTxt: TextView = itemView.findViewById(R.id.jobAppAnnSalTxt)
+        val jobSavedTypeTxt: TextView = itemView.findViewById(R.id.jobTypeTxt)
+        val jobSavedTimeTypeTxt: TextView = itemView.findViewById(R.id.jobTimeTypeTxt)
+        val jobSavedCompanyTxt: TextView = itemView.findViewById(R.id.jobExpCompTxt)
+        val jobSavedRecNameTxt: TextView = itemView.findViewById(R.id.jobExpDesignTxt)
+        val jobSavedAppNameTxt: TextView = itemView.findViewById(R.id.jobAppNameTxt)
+        val jobSavedEmpDescTxt: TextView = itemView.findViewById(R.id.jobEmpDescTxt)
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeSavedJobAdapter.kt:24]: same

File: app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeSavedJobAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        var ItemSavedJob = savedJobList[position]
+        holder.jobSavedTitle.text = ItemSavedJob.jobSavedTitle
+        holder.jobSavedCompanyTxt.text = ItemSavedJob.jobSavedTitle
+        holder.jobSavedAppAnnSalTxt.text = "Rs. "+(ItemSavedJob.jobSavedExMinSalary)+"-"+ItemSavedJob.jobSavedExMaxSalary
+        holder.jobSavedEmpDescTxt.text = ItemSavedJob.jobSavedDescription
+        holder.jobSavedQualification.text =ItemSavedJob.jobSavedQualification
+        holder.jobSavedTypeTxt.text = ItemSavedJob.jobSavedEmpType
+        holder.jobSavedExpSubTxt.text = ItemSavedJob.jobSavedExperience
+        holder.jobSavedTimeTypeTxt.text = ItemSavedJob.jobSavedJobType
+        holder.jobSavedExpSubTxt.text = ItemSavedJob.jobSavedExperience
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/adapters/EmployeeSavedJobAdapter.kt:51]: same

File: app/src/main/java/com/example/teamhiring/presentation/viewmodels/EmployeeSavedJobViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import androidx.lifecycle.ViewModel
+import com.example.teamhiring.data.models.SavedJobData
+import com.example.teamhiring.data.repositories.EmplyeeSavedJobRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import retrofit2.Response
+import javax.inject.Inject
+
+@HiltViewModel
+class EmployeeSavedJobViewModel @Inject constructor(private val savedJobRepository: EmplyeeSavedJobRepository): ViewModel() {
+    suspend fun getSavedJobs(): Response<List<SavedJobData>> = savedJobRepository.getSavedJob()
```
Comment (Nitro2000) [inline app/src/main/java/com/example/teamhiring/presentation/viewmodels/EmployeeSavedJobViewModel.kt:13]: Make one file

**Review by devashish1606 — APPROVED** [review]

**Pull request merged**